base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs

```diff
@@ -91,13 +95,17 @@ where TBackend: OutputManagerBackend + 'static
             })
             .map(
                 |(output, features, script, sender_offset_public_key, metadata_signature)| {
+                    // Todo we need to look here that we might want to fail a specific output and not recover it as this
+                    // will only work if the script is a Nop script. If this is not a Nop script the recovered input
+                    // will not be spendable.
+                    let script_key = PrivateKey::random(&mut OsRng);
```

Comment:
Shouldn't we derive this key from the key manager?

Reply:
We can, but this is in effect a single-use key and can be anything. We can also change this on actual spending of the recovered utxo,  so makes near enough no difference I think.

Reply:
Alright, makes sense